Source/stochastic_internal.h

```diff
@@ -18,7 +18,7 @@ class stochastic_internal : public stochastic {
 
     bool evaluate_probability(double probability) override;
 
-    size_t weight_proportionate_selection(double* weights, size_t num_weights) override;
+    size_t weight_proportionate_selection(const double* const weights, size_t num_weights) override;
```

Comment:
Should this method be marked `const`?

Reply:
This one probably could be marked const. In general the stochastic methods can't be made const because the advance the mersenne twister to the next number in the sequence.

Reply:
Some stochastic-methods cannot be made `const` due to the rng-lib we are using. So this method cannot be marked `const`.